cmake/onnxruntime_common.cmake

```diff
@@ -171,3 +171,73 @@ endif()
 if(APPLE)
   target_link_libraries(onnxruntime_common "-framework Foundation")
 endif()
+
+
+if(MSVC)
+  if(onnxruntime_target_platform STREQUAL "ARM64")
+    set(ARM64 TRUE)
+  elseif (onnxruntime_target_platform STREQUAL "ARM")
+    set(ARM TRUE)
+  elseif(onnxruntime_target_platform STREQUAL "x64")
+    set(X64 TRUE)
+  elseif(onnxruntime_target_platform STREQUAL "x86")
+    set(X86 TRUE)
+  endif()
+elseif(NOT onnxruntime_BUILD_WEBASSEMBLY)
+  if (CMAKE_OSX_ARCHITECTURES STREQUAL "arm64")
+    set(ARM64 TRUE)
+  elseif (CMAKE_OSX_ARCHITECTURES STREQUAL "arm64e")
+    set(ARM64 TRUE)
+  elseif (CMAKE_OSX_ARCHITECTURES STREQUAL "arm")
+    set(ARM TRUE)
+  elseif (CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64")
+    set(X86_64 TRUE)
+  elseif (CMAKE_OSX_ARCHITECTURES STREQUAL "i386")
+    set(X86 TRUE)
+  endif()
+  if (CMAKE_SYSTEM_NAME STREQUAL "Android")
+    if (CMAKE_ANDROID_ARCH_ABI STREQUAL "armeabi-v7a")
+      set(ARM TRUE)
+    elseif (CMAKE_ANDROID_ARCH_ABI STREQUAL "arm64-v8a")
+      set(ARM64 TRUE)
+    elseif (CMAKE_ANDROID_ARCH_ABI STREQUAL "x86_64")
+      set(X86_64 TRUE)
+    elseif (CMAKE_ANDROID_ARCH_ABI STREQUAL "x86")
+      set(X86 TRUE)
+    endif()
+  else()
+    execute_process(
+      COMMAND ${CMAKE_C_COMPILER} -dumpmachine
+      OUTPUT_VARIABLE dumpmachine_output
+      ERROR_QUIET
+    )
+    if(dumpmachine_output MATCHES "^arm64.*")
+      set(ARM64 TRUE)
+    elseif(dumpmachine_output MATCHES "^arm.*")
+      set(ARM TRUE)
+    elseif(dumpmachine_output MATCHES "^aarch64.*")
+      set(ARM64 TRUE)
+    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(i.86|x86?)$")
+      set(X86 TRUE)
+    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(x86_64|amd64)$")
+      set(X86_64 TRUE)
+    endif()
+  endif()
+endif()
```

Comment:
It is dup with mlas.cmake and it doesn't have ARM64EC and power stuff. It'd better to put them in a central place. We can do it later together with a  compiler detection way.